cli/src/semgrep/env.py

```diff
@@ -105,7 +105,12 @@ def version_check_cache_path_default(self) -> Path:
         value = os.getenv("SEMGREP_VERSION_CACHE_PATH")
         if value:
             return Path(value)
-        return Path.home() / ".cache" / "semgrep_version"
+        cache_home = os.getenv("XDG_CACHE_HOME")
+        if cache_home is None or not Path(cache_home).is_dir():
+            parent_dir = Path.home() / ".cache"
+        else:
+            parent_dir = Path(cache_home)
```

Comment:
```suggestion
        if cache_home and Path(cache_home).is_dir():
            parent_dir = Path(cache_home)
        else:
            parent_dir = Path.home() / ".cache"
```
- flip the if-statement for readability
- check for false-ness instead of explicit None to handle case where `XDG_CACHE_HOME` is empty string

Reply:
Gladly!
I've codded it this way only to match the check for `$XDG_CONFIG_HOME` from few lines below.